src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetValue.ts

```diff
@@ -40,7 +40,7 @@ export class DynamicFacetValue implements IDynamicFacetValue {
     this.state = state;
     this.numberOfResults = numberOfResults;
     this.position = position;
-    this.displayValue = displayValue;
+    this.displayValue = displayValue || this.valueCaption;
```

Comment:
A passed down displayValue takes precedence over the valueCaption which is the default.
A Range will pass a formatted displayValue (valueCaption is not an option and is pretty much useless for the DynamicRangeFacet)

Reply:
I understand, my point of view is that the displayValue HAS to be displayed when passed, I don't want to try and get a caption from it (we never know what weird case could show up). The valueCaption is the default fallback if the displayValue is not defined, then if no caption is available the original value will be displayed.

So far there are 3 cases for the displayValue:
1. For the regular DynamicFacet values, no displayValue is passed and it will be defaulted to valueCaption as you would normally expect.
2. For DynamicFacet search, we send the valueCaptions to the API, then get them back directly in the displayValue property of the response, so we shouldn't try to reapply the captions on the captions (maybe there could be some weird bugs there).
3. For the DynamicRangeFacet, we will format the displayValue from the value depending on the format. This facet has no valueCaption option, no not really a factor here. Though it's still unnecessary to try to get them.

Reply:
I see, this is quite tricky because of the differences in responsibilities between the searchapi and searchui as relates to who is in charge of applying the caption.

As a result, this class may be starting to do too much.

One thing we can do to reduce the complexity is to differentiate between `DynamicFacetValues`, `DynamicFacetSearchValues` and `DynamicRangeFacetValues` and apply the appropriate value in each context.

I leave it up to you as to whether or not we should do this at this point. It might still be too early, but worth thinking about nonetheless.

Reply:
What we could do it to extract the valueCaption part outside this class and pass is as a displayValue for the DynamicFacet. The search doesn't use `DynamicFacetValues` and for the ranges we could extend `DynamicFacetValues` in a `DynamicFacetRangeValues` class where only 1 method would need to change.